modelgen/src/test/java/cz/cvut/kbss/jopa/modelgen/OutputFilesGeneratorTests.java

```diff
@@ -0,0 +1,183 @@
+        type6.setTypeName(Set.class.getName());
+        type6.setIsSimple(false);
+        field6.setType(type6);
+        type6.setTypes(Arrays.asList(type1));
+        field6.setType(type6);
+        cl.addField(field6);
+
+        Field field8 = new Field();
+        field8.setAnnotatedWith(Arrays.asList(AnnotationEnum.PROPERTIES));
+        field8.setName("properties");
+        field8.setParentName("cz.test.ex.TestingClass");
+        field8.setType(type6);
+        type6.setTypes(Arrays.asList(type1));
+        field8.setType(type6);
+        cl.addField(field8);
+
+        map.put("cz.test.ex.TestingClass", cl);
+        OutputFilesGenerator.generateOutputFiles(map, "./src/test/java", null, false);
+
+         actualResult = readFileAsString(new File("./src/test/java/cz/test/ex/TestingClass_.java"));
+    }
+
+
+    @Test
+    void isAnnotatedWithSuccessTest() {
+        Field field = new Field();
+        assertFalse(OutputFilesGenerator.isAnnotatedWith(field, null));
+
+        field.setAnnotatedWith(AnnotationEnum.getAll());
+
+        assertTrue(OutputFilesGenerator.isAnnotatedWith(field, AnnotationEnum.ID));
+        assertTrue(OutputFilesGenerator.isAnnotatedWith(field, AnnotationEnum.PROPERTIES));
+        assertTrue(OutputFilesGenerator.isAnnotatedWith(field, AnnotationEnum.TYPES));
+        assertTrue(OutputFilesGenerator.isAnnotatedWith(field, AnnotationEnum.OBJECTPROPERTY));
+        assertTrue(OutputFilesGenerator.isAnnotatedWith(field, AnnotationEnum.ANNOTATIONPROPERTY));
+        assertTrue(OutputFilesGenerator.isAnnotatedWith(field, AnnotationEnum.DATAPROPERTY));
+    }
+
+    @Test
+    void isAnnotatedWithFailTest() {
+        Field field = new Field();
+        assertFalse(OutputFilesGenerator.isAnnotatedWith(field, null));
+
+        field.setAnnotatedWith(Arrays.asList(AnnotationEnum.ID));
+        assertFalse(OutputFilesGenerator.isAnnotatedWith(field, AnnotationEnum.PROPERTIES));
+        assertFalse(OutputFilesGenerator.isAnnotatedWith(field, AnnotationEnum.TYPES));
+        assertFalse(OutputFilesGenerator.isAnnotatedWith(field, AnnotationEnum.OBJECTPROPERTY));
+        assertFalse(OutputFilesGenerator.isAnnotatedWith(field, AnnotationEnum.ANNOTATIONPROPERTY));
+        assertFalse(OutputFilesGenerator.isAnnotatedWith(field, AnnotationEnum.DATAPROPERTY));
+    }
+
+    @Nested
+    class GenerateOutputFiles {
+
+        @AfterEach
+        void deleteSMClass() {
+            File file = new File("./src/test/java/cz/test/ex/TestingClass_.java");
+            file.delete();
+        }
+
+        @Test
+        void containsIdentifier() {
+            assertTrue(actualResult.contains("public static volatile Identifier<TestingClass, URI> uri;"));
+        }
+
+        @Test
+        void containsSingular() {
+            assertTrue(actualResult.contains("public static volatile SingularAttribute<TestingClass, TestingClass> testingClass;"));
+        }
+
+        @Test
+        void containsProperties() {
```

Comment:
Rename to `containsTypedProperties` to avoid confusion with `Map<String, Set<String>> properties`